### PR TITLE
Fix many Kt warnings so as to see the forest for the trees.

### PR DESCRIPTION
**Pull request by shaper**

### CODE REVIEW

File: java/arcs/android/storage/service/BindingContext.kt [hunk 1]
```diff
@@ -66,6 +67,7 @@ class BindingContext(
     private val coroutineContext =
         parentCoroutineContext + job + CoroutineName("BindingContext-$id")
 
+    @ExperimentalCoroutinesApi
```
Comment (jblebrun) [inline java/arcs/android/storage/service/BindingContext.kt:70]: Any reason not to just also mark the entire class? (same question below)

Comment (shaper) [reply java/arcs/android/storage/service/BindingContext.kt]: Done here and for the partition map below. It propagated the annotation to a few add'l places but soon it won't be experimental any more, one hopes, anyway.

**Comment by alxrsngrtn** [comment]
Should we make it so that new Kt warnings break the build?

**Comment by shaper** [comment]
@jblebrun I am going to land this as the conflicts were just a few lines, and I think will merge cleanly.

**Pull request merged**